components/user_settings/advanced/user_settings_advanced.jsx

```diff
@@ -458,6 +510,129 @@ export default class AdvancedSettingsDisplay extends React.PureComponent {
             );
         }
 
+        let smartPasteSection;
+        const {smartPasteTitle, smartPasteDesc} = this.getSmartPasteText();
+        const smartPasteSectionDivider = <div className='divider-light'/>;
+        if (this.props.activeSection === 'advancedSmartPaste') {
+            const smartPasteActive = [
+                this.state.settings.smart_paste === 'true' && this.state.settings.smart_paste_code_blocks === 'true',
+                this.state.settings.smart_paste === 'true' && this.state.settings.smart_paste_code_blocks === 'false',
+                this.state.settings.smart_paste === 'false' && this.state.settings.smart_paste_code_blocks === 'true',
+                this.state.settings.smart_paste === 'false' && this.state.settings.smart_paste_code_blocks === 'false',
+            ];
+
+            const inputs = [
+                <fieldset key='smartPasteSetting'>
+                    <legend className='form-legend hidden-label'>
+                        <FormattedMessage {...smartPasteTitle}/>
+                    </legend>
+                    <div className='radio'>
+                        <label>
+                            <input
+                                id='smartPasteOn'
+                                type='radio'
+                                name='smartPaste'
+                                checked={smartPasteActive[0]}
+                                onChange={() => {
+                                    this.updateSetting('smart_paste', 'true');
+                                    this.updateSetting('smart_paste_code_blocks', 'true');
+                                }}
+                            />
+                            <FormattedMessage
+                                id='user.settings.advance.smartPasteOn'
+                                defaultMessage='Enable smart paste feature'
+                            />
+                        </label>
+                        <br/>
+                    </div>
+                    <div className='radio'>
+                        <label>
+                            <input
+                                id='smartPasteOnWithoutCodeBlocks'
+                                type='radio'
+                                name='smartPaste'
+                                checked={smartPasteActive[1]}
+                                onChange={() => {
+                                    this.updateSetting('smart_paste', 'true');
+                                    this.updateSetting('smart_paste_code_blocks', 'false');
+                                }}
+                            />
+                            <FormattedMessage
+                                id='user.settings.advance.smartPasteOnWithoutCodeBlocks'
+                                defaultMessage='Enable smart paste without auto-detection of code blocks'
```

Comment:
@esethna Around here are the tests shown in the account settings screen to configure smart paste, probably would be great if you can review them. (Probably is easier to get a spinwick running and check the texts there).